client/src/components/pages/RobotInteractionCockpit/RobotInteractionCockpit.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect, useCallback, useRef } from 'react';
+import { Layout, Card, Steps, Space, Button, Typography } from 'antd';
+import HeaderNavbar from '../../content/HeaderNavbar/HeaderNavbar';
+import RobotInteractionInputSection from '../../content/RobotInteractionSections/RobotInteractionInputSection';
+import { getAllParametersForRobot } from '../../../api/variableRetrieval';
+import socket from '../../../utils/socket/socketConnections';
+import { upsert } from '../../../utils/attributeAndParamUtils';
+import styles from './RobotInteractionCockpit.module.css';
+import { isRobotExecutable } from '../../../utils/robotExecution';
+
+const { Step } = Steps;
+const { Title } = Typography;
+
+/**
+ * @description Page, where you can interact with a robot and for example enter input
+ * @category Client
+ * @component
+ */
+const RobotInteractionCockpit = (match) => {
+  const { robotId } = match.match.params;
+  const { userId } = match.location.state;
+  const isMounted = useRef(true);
+  const [parameterList, setParameterList] = useState([]);
+  const [currentStep, setCurrentStep] = useState(0);
+
+  /**
+   * @description For each activity of the current robot get the id, the name and all the parameters that require a user input
+   */
+  const getActivityAndParameterInformation = useCallback(() => {
+    getAllParametersForRobot(robotId)
+      .then((response) => response.json())
+      .then((parameterObjects) => {
+        const activityInformationList = [];
+        Array.prototype.forEach.call(parameterObjects, (parameterObject) => {
+          const { activityId } = parameterObject;
+          let activityName = '';
+          let ssot = sessionStorage.getItem('ssotLocal');
+          ssot = JSON.parse(ssot);
+          Array.prototype.forEach.call(ssot.elements, (elem) => {
+            if (elem.id === activityId) {
+              activityName = elem.name;
+            }
+          });
+          const activityParameter = [];
+          Array.prototype.forEach.call(
+            parameterObject.rpaParameters,
```

Comment:
What happens if there are no such parameters set?

Reply:
That is not a problem. In this case just nothing would be added to activityParameter so it would be an empty array. In the component RobotInteractionInputSection in line 19 we are checking if the array activityParameter is empty and just render new parameter input fields if it is not empty.